### PR TITLE
[google_maps_flutter] Fix ios tile overlay not loading when dealing with wide gamut images

### PR DESCRIPTION
**Pull request by LouiseHsu**
Fixes https://github.com/flutter/flutter/issues/135252
Google maps does not currently support the use of wide gamut images.
Since engine uses 16 bit floats when dealing with wide gamut images ([here](https://github.com/flutter/engine/blob/8a51e124fbf168295a1b05f8d66459bfb29ae8a9/shell/platform/darwin/ios/framework/Source/FlutterView.mm#L125)), when we detect that an image is using that format, we downsample it.

https://github.com/flutter/packages/assets/36148254/2d67418d-53a9-472b-b03e-86fb24f9610c

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/google_maps_flutter/google_maps_flutter_ios/example/ios14/ios/Runner.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -589,13 +652,74 @@
...
+			name = Release;
+		};
 		F7151F17265D7ED70028CB91 /* Debug */ = {
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = E52C6A6210A56F027C582EF9 /* Pods-RunnerTests.debug.xcconfig */;
 			buildSettings = {
 				BUNDLE_LOADER = "$(TEST_HOST)";
 				CLANG_ENABLE_MODULES = YES;
 				CODE_SIGN_STYLE = Automatic;
+				DEVELOPMENT_TEAM = S8QB4VV633;
```
Comment (hellohuanlin) [inline packages/google_maps_flutter/google_maps_flutter_ios/example/ios14/ios/Runner.xcodeproj/project.pbxproj:722]: revert

File: packages/google_maps_flutter/google_maps_flutter_ios/example/ios14/ios/RunnerTests/GoogleMapsTests.m [hunk 2]
```diff
@@ -62,4 +66,20 @@ - (void)testMapsServiceSync {
   XCTAssertEqual(factory1.sharedMapServices, factory2.sharedMapServices);
 }
 
+- (void)testWideGamutImagesProperlyHandled {
```
Comment (hellohuanlin) [inline packages/google_maps_flutter/google_maps_flutter_ios/example/ios14/ios/RunnerTests/GoogleMapsTests.m:69]: nit: `test[FunctionNameBeingTested][WhatsBeingTested]`

File: packages/google_maps_flutter/google_maps_flutter_ios/example/ios14/ios/RunnerTests/GoogleMapsTests.m [hunk 3]
```diff
@@ -62,4 +66,20 @@ - (void)testMapsServiceSync {
...
+  FLTTileProviderController *controller = [[FLTTileProviderController alloc] init];
+
+  UIImage *wideGamutImage = [UIImage imageWithContentsOfFile:@"assets/widegamut.png"];
+
+  UIImage *downsampledImage = [controller handleResultTile:wideGamutImage];
+
+  CGImageRef imageRef = downsampledImage.CGImage;
+  CGBitmapInfo bitmapInfo = CGImageGetBitmapInfo(imageRef);
+  bool isFloat = (bitmapInfo && kCGBitmapFloatComponents);
```
Comment (hellohuanlin) [inline packages/google_maps_flutter/google_maps_flutter_ios/example/ios14/ios/RunnerTests/GoogleMapsTests.m:78]: `BOOL`

Comment (hellohuanlin) [reply packages/google_maps_flutter/google_maps_flutter_ios/example/ios14/ios/RunnerTests/GoogleMapsTests.m:78]: remove `()`

File: packages/google_maps_flutter/google_maps_flutter_ios/example/ios14/ios/RunnerTests/GoogleMapsTests.m [hunk 4]
```diff
@@ -62,4 +66,20 @@ - (void)testMapsServiceSync {
...
+  size_t bitsPerComponent = CGImageGetBitsPerComponent(imageRef);
+
+  XCTAssert(isFloat == false);
```
Comment (hellohuanlin) [inline packages/google_maps_flutter/google_maps_flutter_ios/example/ios14/ios/RunnerTests/GoogleMapsTests.m:81]: `XCTAssertFalse`

File: packages/google_maps_flutter/google_maps_flutter_ios/example/ios14/ios/RunnerTests/GoogleMapsTests.m [hunk 5]
```diff
@@ -62,4 +66,20 @@ - (void)testMapsServiceSync {
...
+  XCTAssert(bitsPerComponent != 16);
```
Comment (hellohuanlin) [inline packages/google_maps_flutter/google_maps_flutter_ios/example/ios14/ios/RunnerTests/GoogleMapsTests.m:82]: what's the value actually gonna be?

File: packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/FLTGoogleMapTileOverlayController.m [hunk 6]
```diff
@@ -118,6 +118,31 @@ - (instancetype)init:(FlutterMethodChannel *)methodChannel
 
 #pragma mark - GMSTileLayer method
 
+- (UIImage *)handleResultTile:(nullable UIImage*)tile {
+    CGImageRef imageRef = tile.CGImage;
+    CGBitmapInfo bitmapInfo = CGImageGetBitmapInfo(imageRef);
+    bool isFloat = (bitmapInfo && kCGBitmapFloatComponents);
```
Comment (hellohuanlin) [inline packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/FLTGoogleMapTileOverlayController.m:124]: same here

File: packages/google_maps_flutter/google_maps_flutter_ios/example/ios14/ios/RunnerTests/GoogleMapsTests.m (hunk 5 above)
Comment (LouiseHsu) [reply packages/google_maps_flutter/google_maps_flutter_ios/example/ios14/ios/RunnerTests/GoogleMapsTests.m:82]: non-wide gamut imgs are 8 bit float format

**Pull request merged**